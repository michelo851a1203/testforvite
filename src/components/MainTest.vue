<script setup lang="ts">
  import { ref, computed } from 'vue';
  const props = withDefaults(defineProps<{
    buttonTitle: string;
    timesNumber: number;
  }>(), {
    buttonTitle: 'click',
    timesNumber: 1,
  });

  const emit = defineEmits<{
    (e: 'update:callbackClick', inputNumber: number): void
  }>()

  const inputNumber = ref(0);

  const clickHandler = () => {
    inputNumber.value++;
    emit('update:callbackClick', inputNumber.value);
  }

  const resultNumber = computed(() => {
    return props.timesNumber * inputNumber.value;
  });
</script>

<template>
  <button
    @click="clickHandler"
    class="border-none bg-blue-300 rounded-md px-4 py-3"
  >
    {{ buttonTitle }} - {{ resultNumber }}
  </button>
</template>
