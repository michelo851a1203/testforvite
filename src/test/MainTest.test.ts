import { mount, VueWrapper } from '@vue/test-utils';
import MainTest from '../components/MainTest.vue';

test('mount component', async () => {
  const buttonTitle = 'testClickHello';
  const timesNumber = 2.5;
  const initialInputNumber = 0;
  const inputTitleInitialResult = `${buttonTitle} - ${timesNumber * initialInputNumber}`

  const wrapper = mount(MainTest, {
    props: {
      buttonTitle,
      timesNumber,
    },
  });

  expect(wrapper.text()).toContain(inputTitleInitialResult);
  expect(wrapper.html()).toMatchSnapshot();
  await wrapper.get('button').trigger('click');
  expect(wrapper.text()).toContain(`${buttonTitle} - ${timesNumber * (initialInputNumber + 1)}`);
  testEmitWithWrapper(wrapper, 1);
});

const testEmitWithWrapper = (
  wrapper: VueWrapper<any>,
  expectNumber: number,
) => {
  const wrapperEmit = wrapper.emitted()['update:callbackClick'];
  const lastIndex = wrapperEmit.length - 1;
  expect(wrapperEmit).toBeTruthy();
  expect(wrapperEmit[lastIndex]).toEqual([expectNumber]);
}

