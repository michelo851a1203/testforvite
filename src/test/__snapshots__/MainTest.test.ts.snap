// Vitest Snapshot v1

exports[`mount component 1`] = `"<button class=\\"border-none bg-blue-300 rounded-md px-4 py-3\\">testClickHello - 0</button>"`;
